rts/at/integration/src/main/java/org/jboss/narayana/rest/integration/RecoveryManager.java

```diff
@@ -93,12 +100,15 @@ public void removeParticipantInformation(final ParticipantInformation participan
         }
 
         final RecoveryStore recoveryStore = StoreManager.getRecoveryStore();
-        final Uid uid = new Uid(participantInformation.getId());
+        String participantId = participantInformation.getId();
 
-        try {
-            recoveryStore.remove_committed(uid, PARTICIPANT_INFORMATION_RECORD_TYPE);
-        } catch (ObjectStoreException e) {
-            LOG.warn("Failure while removing participant information from the object store.", e);
+        if(persistedParticipants.get(participantId) != null) {
```

Comment:
@ochaloup is there any need to synchronise in case a thread doing write_participant and persistedParticipants.put could execute concurrently with a thread checking if something is in the map? I don't know the answer but did you assess that?

Reply:
yes, I did. There is used the `ConcurrentHashMap`. If something is checking (e.g. recovery) if something is in map, it may not find it but it will find it in the next cycle.